src/components/time-picker/time-picker.js

```diff
@@ -0,0 +1,96 @@
+import mixin from '../../globals/js/misc/mixin';
+import createComponent from '../../globals/js/mixins/create-component';
+import initComponentBySearch from '../../globals/js/mixins/init-component-by-search';
+
+class TimePicker extends mixin(createComponent, initComponentBySearch) {
+  /**
+   * Time picker UI.
+   * @extends CreateComponent
+   * @extends InitComponentBySearch
+   * @param {HTMLElement} element The element working as a time picker UI.
+   */
+  constructor(element, options) {
+    super(element, options);
+    this._initInputEvents();
+    this._setMinMaxValues();
+    this.element.militaryTime = false;
+  }
+
+  setMilitaryTime = () => {
+    this.element.militaryTime = true;
+    this._setMinMaxValues();
+  }
+
+  _setMinMaxValues = () => {
+    const hourInput = this.element.querySelector(this.options.classHoursInput);
+    if (this.element.militaryTime) {
+      hourInput.max = 24;
+    } else {
+      hourInput.max = 12;
+    }
+  }
+
+  _initInputEvents = () => {
+    [...this.element.querySelectorAll(this.options.selectorTimePickerInputField)].forEach((input) => {
+      const curInput = input;
+      const isText = (input.type === 'text');
+      let lastKey;
+      input.addEventListener('focus', () => {
+        this.element.querySelector(this.options.selectorTimePickerInput).classList.add('focused');
+        input.select();
+      });
+      input.addEventListener('blur', () => {
+        this.element.querySelector(this.options.selectorTimePickerInput).classList.remove('focused');
+      });
+      input.addEventListener('change', () => {
+        if (curInput.value.length === 1 && !isText) {
+          curInput.value = `0${curInput.value}`;
+        }
+      });
+      input.addEventListener('keydown', (e) => {
+        lastKey = e.which;
+      });
+      input.addEventListener('input', () => {
+        if (isText) {
+          if (curInput.value === 'a' || curInput.value === 'A') {
```

Comment:
Maybe you can do `curInput.value.toUpperCase()` before you do the conditionals so that you only need to check the letter and not the casing.